drivers/usb/device/usb_dc_mcux.c

```diff
@@ -503,19 +515,38 @@ int usb_dc_ep_write(const uint8_t ep, const uint8_t *const data,
 		return -EINVAL;
 	}
 
-#if defined(CONFIG_HAS_MCUX_CACHE)
-	DCACHE_CleanByRange((uint32_t)data, data_len);
+	/* Copy the data for SoC's that do not have a USB RAM
+	 * as the SDK driver will copy the data into USB RAM,
+	 * if available.
+	 */
+#ifndef FSL_FEATURE_USBHSD_USB_RAM_BASE_ADDRESS
+	buffer = (uint8_t *)dev_state.eps[ep_abs_idx].block.data;
+
+	if (data_len > CONFIG_USB_MCUX_ENDPOINT_BUFFER_SIZE) {
+		LOG_ERR("Packet too large");
```

Comment:
The buffer is allocated for a size of `cfg->ep_mps`. However the check is done against `CONFIG_USB_MCUX_ENDPOINT_BUFFER_SIZE`. We probably don't need this Kconfig

Reply:
yes I had this on my local branch, I forgot to push